srcpkgs/sile/patches/upstream-cross.patch

```diff
@@ -0,0 +1,64 @@
+From 19c7c1d4cf07ac7baa7c4c6b24d0d82f2dcea0af Mon Sep 17 00:00:00 2001
+From: Caleb Maclennan <caleb@alerque.com>
+Date: Mon, 25 Nov 2024 12:27:25 +0300
+Subject: [PATCH] fix(build): Support cross-compilation of Rust binaries
+ (#2178)
+
+---
+ Makefile.am                       | 2 +-
+ build-aux/que_rust_boilerplate.am | 4 ++--
+ build-aux/que_rust_boilerplate.m4 | 8 ++++++--
+ 3 files changed, 9 insertions(+), 5 deletions(-)
+
+diff --git a/Makefile.am b/Makefile.am
+index a5fc43102..bb44bdb5e 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -179,7 +179,7 @@ CARGO_FEATURE_ARGS += --features variations
+ endif
+ 
+ rusile.so: $(rusile_so_SOURCES) $(bin_PROGRAMS)
+-	$(CARGO_ENV) $(CARGO) build $(CARGO_VERBOSE) $(RUSILE_FEATURE_ARG) $(CARGO_RELEASE_ARGS) -p rusile
++	$(CARGO_ENV) $(CARGO) build $(CARGO_VERBOSE) --target $(CARGO_TARGET_TRIPLE) $(RUSILE_FEATURE_ARG) $(CARGO_RELEASE_ARGS) -p rusile
+ 	$(INSTALL) @builddir@/target/@RUST_TARGET_SUBDIR@/lib$@ $@
+ 
+ DEPDIR := .deps
+diff --git a/build-aux/que_rust_boilerplate.am b/build-aux/que_rust_boilerplate.am
+index a1ba65af4..26de4ef58 100644
+--- a/build-aux/que_rust_boilerplate.am
++++ b/build-aux/que_rust_boilerplate.am
+@@ -57,8 +57,8 @@ $(COMPLETIONS_OUT_DIR)/_$(TRANSFORMED_PACKAGE_NAME): $(CARGO_BIN) | $(COMPLETION
+ $(_RUST_OUT) $(CARGO_BIN): $(@PACKAGE_VAR@_SOURCES) $(nodist_@PACKAGE_VAR@_SOURCES) $(EXTRA_@PACKAGE_VAR@_SOURCES)
+ 	set -e
+ 	export AUTOTOOLS_DEPENDENCIES="$^"
+-	$(CARGO_ENV) $(CARGO) build $(CARGO_VERBOSE) $(CARGO_FEATURE_ARGS) $(CARGO_RELEASE_ARGS)
+-	$(CARGO_ENV) $(CARGO) build --quiet --message-format=json $(CARGO_FEATURE_ARGS) $(CARGO_RELEASE_ARGS) | \
++	$(CARGO_ENV) $(CARGO) build $(CARGO_VERBOSE) --target $(CARGO_TARGET_TRIPLE) $(CARGO_FEATURE_ARGS) $(CARGO_RELEASE_ARGS)
++	$(CARGO_ENV) $(CARGO) build  --target $(CARGO_TARGET_TRIPLE) --quiet --message-format=json $(CARGO_FEATURE_ARGS) $(CARGO_RELEASE_ARGS) | \
+ 		$(JQ) -sr 'map(select(.reason == "build-script-executed")) | last | .out_dir' > $(_RUST_OUT)
+ 
+ RUST_DEVELOPER_TARGETS = cargo-test clippy rustfmt
+diff --git a/build-aux/que_rust_boilerplate.m4 b/build-aux/que_rust_boilerplate.m4
```

Comment:
Yes, somehow the automatic GCC/LD build stuff is trying to take over the Rust-built binary. That's not supposed to happen. I do see to recall something like this turning up before, but I can't remember what the mitigation was.

Reply:
I see this patch is getting applied, but it is not being *used* because after patching the build moves straight on to running `./configure` which is just the pre-generated one that comes in the tarball. Since the patch updates the *template* and *macros, the configure script needs to be regenerated before use to actually reflect the changes in the patch.

`autoreconf -fi` should do the trick, run after patching and before configure starts:

```
2024-11-26T08:55:11.3883538Z sile-0.15.6.tar.zst: 3254KB [avg rate: 26GB/s]
2024-11-26T08:55:11.3999482Z [97m[1m=> sile-0.15.6_1: verifying checksum for distfile 'sile-0.15.6.tar.zst'... [m[97m[1mOK.
2024-11-26T08:55:11.4303411Z [m[97m[1m=> sile-0.15.6_1: running do-extract hook: 00-distfiles ...
2024-11-26T08:55:11.4343450Z [m[97m[1m=> sile-0.15.6_1: extracting distfile(s), please wait...
2024-11-26T08:55:11.5958461Z [m[97m[1m=> sile-0.15.6_1: running do-patch hook: 00-patches ...
2024-11-26T08:55:11.5993810Z [m[97m[1m=> sile-0.15.6_1: patching: upstream-cross.patch.
2024-11-26T08:55:11.6318198Z [m[97m[1m=> sile-0.15.6_1: running pre-configure hook: 00-gnu-configure-asneeded ...
2024-11-26T08:55:11.6441305Z [m[97m[1m=> sile-0.15.6_1: running pre-configure hook: 01-override-config ...
2024-11-26T08:55:11.6699671Z [m[97m[1m=> sile-0.15.6_1: running pre-configure hook: 02-script-wrapper ...
2024-11-26T08:55:11.6810168Z [m[97m[1m=> sile-0.15.6_1: running do_configure ...
2024-11-26T08:55:11.7922483Z [mchecking for a BSD-compatible install... /builddir/.xbps-sile/wrappers/install -c
2024-11-26T08:55:11.7950440Z checking whether build environment is sane... yes
2024-11-26T08:55:11.8026031Z checking for a race-free mkdir -p... /usr/bin/mkdir -p
```

This log suggests there might be some pre-configure hook you can utilize to trigger autoreconf, or there might even be a premade hook just for that since it is a common operation to need to run on autotools builds, but I don't know the ins and outs of Void packaging scripts.